src/journal.rs

```diff
@@ -66,7 +60,8 @@ pub type JournalRecord = BTreeMap<String, String>;
 ///
 /// Supports read, next, previous, and seek operations.
 pub struct Journal {
-    j: ffi::sd_journal
+    j: ffi::sd_journal,
+    cursor: String,
```

Comment:
I'm using this for iterator. I'll create a function (wrapper) which gets current cursor and use it if you prefer it that way